File: src/TripleTriad.Client/src/app/games/models/card.ts
export interface Card {
  name: string;
  level: number;
  rank: Rank;
  element: string;
}

export interface Rank {
  top: number;
  right: number;
  bottom: number;
  left: number;
}

export interface CardListCard extends Card {
  canSelect: boolean;
}

export interface SelectedCardListCard extends Card {
  canRemove: boolean;
}

File: src/TripleTriad.Logic/Entities/Card.cs
using TripleTriad.Logic.Enums;

namespace TripleTriad.Logic.Entities
{
    public class Card
    {
        public Card(string name, int level, Rank rank, Element? element = null)
        {
            Name = name;
            Level = level;
            Rank = rank;
            Element = element;
        }

        public string Name { get; }

        public int Level { get; }

        public Rank Rank { get; }

        public Element? Element { get; }
    }
}